docs/samples/Microsoft.ML.Samples/Dynamic/LightGBM/LightGBMBinaryClassification.cs

```diff
@@ -0,0 +1,103 @@
+                    new TextLoader.Column("capital-loss", DataKind.R4, 11),
+                    new TextLoader.Column("hours-per-week", DataKind.R4, 12),
+                    new TextLoader.Column("native-country", DataKind.Text, 13),
+                    new TextLoader.Column("Label", DataKind.Bool, 14)
+                }
+            });
+
+            // Read the data, and leave 10% out, so we can use them for testing
+            var (trainData, testData) = mlContext.BinaryClassification.TrainTestSplit(reader, testFraction: 0.1);
+
+            // Create the Estimator
+            var pipeline = mlContext.Transforms.Categorical.OneHotEncoding(new OneHotEncodingEstimator.ColumnInfo[]
+                {
+                    new OneHotEncodingEstimator.ColumnInfo("marital-status"),
+                    new OneHotEncodingEstimator.ColumnInfo("occupation"),
+                    new OneHotEncodingEstimator.ColumnInfo("relationship"),
+                    new OneHotEncodingEstimator.ColumnInfo("ethnicity"),
+                    new OneHotEncodingEstimator.ColumnInfo("sex"),
+                    new OneHotEncodingEstimator.ColumnInfo("native-country"),
+                })
+            .Append(mlContext.Transforms.FeatureSelection.SelectFeaturesBasedOnCount("native-country", count: 10))
+            .Append(mlContext.Transforms.Concatenate("Features",
+                                                        "age",
+                                                        "education-num",
+                                                        "marital-status",
+                                                        "relationship",
+                                                        "ethnicity",
+                                                        "sex",
+                                                        "hours-per-week",
+                                                        "native-country"))
+            .Append(mlContext.Transforms.Normalize("Features"))
+            .Append(mlContext.BinaryClassification.Trainers.LightGbm());
+
+            // Fit this Pipeline to the Training Data
+            var model = pipeline.Fit(trainData);
+
+            // Evaluate how the model is doing on the test data
+            var dataWithPredictions = model.Transform(testData);
+
+            var metrics = mlContext.BinaryClassification.Evaluate(dataWithPredictions);
+
+            Console.WriteLine($"Accuracy: {metrics.Accuracy}"); // 0.84
+            Console.WriteLine($"AUC: {metrics.Auc}"); // 0.88
+            Console.WriteLine($"F1 Score: {metrics.F1Score}"); // 0.62
+
+            Console.WriteLine($"Negative Precision: {metrics.NegativePrecision}"); // 0.88
+            Console.WriteLine($"Negative Recall: {metrics.NegativeRecall}"); // 0.91
+            Console.WriteLine($"Positive Precision: {metrics.PositivePrecision}"); // 0.67
+            Console.WriteLine($"Positive Recall: {metrics.PositiveRecall}"); // 0.58       
```

Comment:
see : [https://github.com/dotnet/machinelearning/pull/2483](https://github.com/dotnet/machinelearning/pull/2483)

Reply:
To be consistent, I am going to change it the way [https://github.com/dotnet/machinelearning/pull/2483](https://github.com/dotnet/machinelearning/pull/2483) is using it. 

However, I think that this sort of code reusability does not make these sample standalone which I think is more important for documentation samples. What do you think?

---
In reply to: [255272930](https://github.com/dotnet/machinelearning/pull/2476#discussion_r255272930) [](ancestors = 255272930)